src/measure_theory/outer_measure.lean

```diff
@@ -353,6 +355,47 @@ theorem le_of_function {μ : outer_measure α} :
 
 end of_function
 
+section bounded_by
+
+variables {α : Type*} (m : set α → ennreal)
+
+/-- Given any function `m` assigning measures to sets, there is a unique maximal outer measure `μ`
+  satisfying `μ s ≤ m s` for all `s : set α`. This is the same as `outer_measure.of_function`,
+  except that it doesn't require `m ∅ = 0`. -/
+def bounded_by : outer_measure α :=
+outer_measure.of_function (λ s, ⨆ (h : s.nonempty), m s) (by simp [empty_not_nonempty])
```

Comment:
Is `⨆ (h : s.nonempty), m s` more convenient that `if s.nonempty then m s else 0` or `function.update m ∅ 0`?

Reply:
Anyway,
bors d+

Reply:
I think the answer is "no". I'm just using what was used before in the measure theory library.
I think `⨆ (h : s.nonempty), _` is better if you don't use classical logic (since it doesn't require any decidability), but we moved away from that anyway. I'm happy to change it to `function.update` at some point.